tests/pytorch/test_optim.py

```diff
@@ -42,5 +45,173 @@ def test_sparse_adam():
     # Pytorch sparseAdam maintains a global step
     # DGL sparseAdam use a per embedding step
 
+@unittest.skipIf(os.name == 'nt', reason='Do not support windows yet')
+def test_sparse_adam_zero_step():
+    num_embs = 10
+    emb_dim = 4
+    device=F.ctx()
+    dgl_emb = NodeEmbedding(num_embs, emb_dim, 'test')
+    torch_emb = th.nn.Embedding(num_embs, emb_dim, sparse=True)
+    dgl_emb_zero = NodeEmbedding(num_embs, emb_dim, 'test2')
+    torch_emb_zero = th.nn.Embedding(num_embs, emb_dim, sparse=True)
+    th.manual_seed(0)
+    th.nn.init.uniform_(torch_emb.weight, 0, 1.0)
+    th.nn.init.uniform_(torch_emb_zero.weight, 0, 1.0)
+    th.manual_seed(0)
+    th.nn.init.uniform_(dgl_emb.emb_tensor, 0, 1.0)
+    th.nn.init.uniform_(dgl_emb_zero.emb_tensor, 0, 1.0)
+
+    dgl_adam = SparseAdam(params=[dgl_emb, dgl_emb_zero], lr=0.01)
+    torch_adam = th.optim.SparseAdam(
+        list(torch_emb.parameters()) + list(torch_emb_zero.parameters()), lr=0.01)
+
+    # first step
+    idx = th.randint(0, num_embs, size=(4,))
+    dgl_value = dgl_emb(idx, device).to(th.device('cpu'))
+    torch_value = torch_emb(idx)
+    labels = th.ones((4,)).long()
+
+    dgl_adam.zero_grad()
+    torch_adam.zero_grad()
+    dgl_loss = th.nn.functional.cross_entropy(dgl_value, labels)
+    torch_loss = th.nn.functional.cross_entropy(torch_value, labels)
+    dgl_loss.backward()
+    torch_loss.backward()
+
+    dgl_adam.step()
+    torch_adam.step()
+    assert F.allclose(dgl_emb.emb_tensor, torch_emb.weight)
```

Comment:
should we rename `emb_tensor` to `weight` as well?

Reply:
Good point.